services/web/server/tests/unit/with_dbs/01/test_comp_tasks_listening_task.py

```diff
@@ -138,11 +118,22 @@ async def test_listen_comp_tasks_task(
         # let's update some values
         await conn.execute(
             comp_tasks.update()
-            .values(**upd_value)
+            .values(**update_values)
             .where(comp_tasks.c.task_id == task["task_id"])
         )
-        for key, mock_fct in mock_project_subsystem.items():
-            if key in exp_calls:
-                await _wait_for_call(mock_fct)
+
+        # tests whether listener gets hooked calls executed
+        for call_name, mocked_call in mock_project_subsystem.items():
+            if call_name in expected_calls:
+                # async.sleep if and retry if not called
+                async for attempt in tenacity.AsyncRetrying(
+                    wait=tenacity.wait_fixed(1),
+                    stop=tenacity.stop_after_delay(10),
+                    retry=tenacity.retry_if_exception_type(AssertionError),
+                    before=tenacity.before_log(logger, logging.INFO),
+                    reraise=True,
+                ):
+                    with attempt:
```

Comment:
not sure I find that better/more readable than the function with a decorator. was it required for 3.8? also less reusable.

Reply:
yeah, it is a bit bulky ... I simplified it